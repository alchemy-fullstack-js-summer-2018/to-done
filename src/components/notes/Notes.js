import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class Notes extends Component {

  static propTypes = {
    notes: 
    PropTypes.array,
    onUpdate: PropTypes.func.isRequired,
    onRemove: PropTypes.func.isRequired
  };

  render() {
    const { notes, onUpdate, onRemove } = this.props;

    return (
      <ul>
        {notes.map(note => (
          <Note
            key={note.key}
            note={note}
            onUpdate={onUpdate}
            onRemove={onRemove}
          />
        ))}
      </ul>
    );
  }
}
export default Notes;