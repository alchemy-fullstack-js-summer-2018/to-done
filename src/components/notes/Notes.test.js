import React from 'react';
import { shallow } from 'enzyme';
import Notes from './Notes';
import toJSON from 'enzyme-to-json';

describe('Notes test', () => {
  
  it('renders add if no note prop', () => {
  
    const notes = [
      { title: 'Nissan', content: 'Car Bio', key: 'abc' },
      { title: 'Top Gear', content: 'Show', key: 'def' },
      { title: 'KOF', content: 'Video Game', key: 'ghi' },
    ];
    const wrapper = shallow(<Notes 
      notes={notes}
      onUpdate={jest.fn()}
      onRemove={jest.fn()}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});