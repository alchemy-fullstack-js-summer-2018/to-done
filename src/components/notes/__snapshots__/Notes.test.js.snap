// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes test renders add if no note prop 1`] = `
<ul
  className="notes-container"
>
  <Note
    key="abc"
    note={
      Object {
        "content": "Car Bio",
        "key": "abc",
        "title": "Nissan",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="def"
    note={
      Object {
        "content": "Show",
        "key": "def",
        "title": "Top Gear",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="ghi"
    note={
      Object {
        "content": "Video Game",
        "key": "ghi",
        "title": "KOF",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
</ul>
`;
